### PR TITLE
feat: add cancel transaction, ref #LEA-958

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 53a7e31 — [Extension build](https://github.com/leather-io/extension/actions/runs/11592147513), [Test report](https://leather-io.github.io/playwright-reports/feat-cancel-tx), [Storybook](https://feat-cancel-tx--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat-cancel-tx)

This pr adds option to cancel stx transaction, also adds tests for increase and cancel stx transaction

@Aman-zishan thanks a lot for your work on this

### CODE REVIEW

File: src/app/components/stacks-transaction-item/stacks-transaction-action-menu.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { ActivitySelectors } from '@tests/selectors/activity.selectors';
+import { css } from 'leather-styles/css';
+import { HStack, styled } from 'leather-styles/jsx';
+
+import { ChevronDownIcon, ChevronsRightIcon, CloseIcon, DropdownMenu, Flag } from '@leather.io/ui';
+
+interface Props {
```
Comment (kyranjamie) [inline src/app/components/stacks-transaction-item/stacks-transaction-action-menu.tsx]: use full name like other components 🙏🏼

File: src/app/components/stacks-transaction-item/stacks-transaction-item.tsx [hunk 2]
```diff
@@ -58,10 +61,21 @@ export function StacksTransactionItem({
...
+  const txIcon = transaction ? <StacksTransactionIcon transaction={transaction} /> : icon;
+  const txTitle = transaction ? getTxTitle(transaction) : title || '';
+  const txValue = transaction ? getTxValue(transaction, isOriginator) : value;
+
+  function handleTransactionAction(action: 'cancel' | 'increaseFee') {
     if (!transaction) return;
 
-    const routeUrl = RouteUrls.IncreaseStxFee.replace(':txid', transaction.tx_id);
+    const routeUrl =
+      action === 'increaseFee'
```
Comment (fbwoolf) [inline src/app/components/stacks-transaction-item/stacks-transaction-item.tsx]: Maybe pull this out as a defined type and use `increase-fee`?

**Review by fbwoolf — APPROVED** [review]

File: src/shared/route-urls.ts [hunk 3]
```diff
@@ -26,8 +26,9 @@ export enum RouteUrls {
   EditNetwork = '/edit-network',
   Fund = '/fund/:currency',
   FundChooseCurrency = '/fund-choose-currency',
-  IncreaseStxFee = '/increase-fee/stx/:txid',
+  IncreaseStacksFee = '/increase-fee/stacks/:txid',
```
Comment (pete-watters) [inline src/shared/route-urls.ts:29]: I guess `/increase-fee/btc` should also update here?

**Comment by pete-watters** [comment]
This code looks good but I am confused about how to cancel a transaction. 

I tried to cancel a stacks transfer and I see this:
Screenshot 2024-10-29 at 13 36 13

Do I need to choose a fee and then submit to cancel? 🤔 So you make a new reversed transaction?

**Review by pete-watters — APPROVED** [review]
👍

**Comment by alter-eggo** [comment]
@314159265359879 could you test this build please

**Comment by alter-eggo** [comment]
@pete-watters yeah, as I understand you need to increase fee a bit to cancel tx

**Comment by 314159265359879** [comment]
Thanks for this @alter-eggo 
The prefilled fee here should default to 0.000001 STX more than the current fee for the transaction being replaced. That way the user can already click submit (now it is not allowed unless they click atleast 2x, which will be excessive in most cases).
With a 0.000001 STX increase the "submit" button can be light up right away (ready to click, and then increasing the fee further is optional).

image

For context, STX transfers don't usually need a higher fee to get included in a block (unlike some contract calls and contract deployments) they require very little of the "block space".  

image

With the change above. I opt to remove "A higher fee can help replace the old transaction." from the explainer. Because in most cases we'll set the right fee (just 0.000001 STX more then current). Leaving it in there may make users think that they have to set a much higher fee, which is not actually the case. There is always still a chance the old transaction gets processed anyway, because it reaches a miner first or before it is replaced by the new transaction with the higher fee. 

"To cancel the transaction we replace it with a minimal STX transfer." Could be a good add here so users are not surprised that the "cancelled" transaction goes away and a small STX transfer is displayed in its place. 

I think this is out of scope now: Later we can create an enhancement to label the transaction differently (based on the memo) used in the cancel transaction. "Pending Cancellation" instead of the normal "Pending".

**Review by 314159265359879 — CHANGES_REQUESTED** [review]
See comments above.

File: src/app/components/stacks-transaction-item/stacks-transaction-item.tsx (hunk 2 above)
Comment (alter-eggo) [reply src/app/components/stacks-transaction-item/stacks-transaction-item.tsx]: added enum

File: src/shared/route-urls.ts (hunk 3 above)
Comment (alter-eggo) [reply src/shared/route-urls.ts:29]: how `/increase-fee/btc` should be changed here? 🤔

Comment (pete-watters) [reply src/shared/route-urls.ts:29]: I just saw you changed:
- `IncreaseStxFee` -> `IncreaseStacksFee` 
- `'/increase-fee/stx/:txid'` -> `/increase-fee/stacks/:txid`

Should we also have `IncreaseBitcoinFee`  & ``/increase-fee/bitcoin/:txid` then?

Comment (alter-eggo) [reply src/shared/route-urls.ts:29]: no, we don't need, we pass tx in state for btc. I haven't touched btc increase flow in this pr

Comment (pete-watters) [reply src/shared/route-urls.ts:29]: OK 👍 I was just suggesting for consistency as there is a mix of conventions here

**Comment by alter-eggo** [comment]
thanks @314159265359879 fixed this

**Review by 314159265359879 — APPROVED** [review]
Looking great!

**Pull request merged**